### DISCUSSION TITLE
How can I search without filtering

### DISCUSSION THREAD

**Discussion by PhilJollans** [datalust/seq-tickets#2275] [Help] [answered]
I know that I can enter a search expression and that Seq will filter the log events.  
Is there a way to search the log events without applying a filter?

I want to find a particular log entry, for example a REST call, and then look at what happens immediately after this event. I can find the event in the viewer, but it filters out the following events. Currently I am searching for an event, noting down its exact timestamp, removing the filter expression and then scrolling down to find the timestamp. I can't believe that this is the right approach.

Phil

**Comment by liammclennan** [comment, marked as answer]
Hi @PhilJollans 
Once you find your event you can choose expand it and choose _Event_ > _Seek to +/- 5s_, then clear the filter to see all events. 

Its also common to find an event and then search by some correlation id, or trace id to find related events.

**Comment by PhilJollans** [comment]
Ok, that certainly helps.
I kind of have the feeling that a "search and highlight " feature is missing, but I'm still learning how to use Seq :wink:.

**Comment by PhilJollans** [comment]
Since my original question, I have used seq a lot more.
This is still a pain point for me. I would really like the ability to:
 - search for the next match to an expression, without filtering the results
 - search and highlight all matches in a given time span
 - bookmark a log entry

However, I don't think I am a typical user of seq. We are not logging directly to the server, but using the ingest command to load and analyze log files (from multiple sources) in the serilog compact log format. So we are basically using it as a log file viewer.  

This may mean, that something which is a completely obvious feature for me, is impractical in the scenario of a permanently running log server.

**Reply by nblumhardt** [reply]
Thanks for the follow-up, @PhilJollans. "Search within results" is definitely something we'd like to explore further, it's been on the radar for a while and we have a candidate design we've discussed in the past. The approach we'd like to use requires quite a bit of change in _Events_ so it's waiting for an opportune moment :-)
